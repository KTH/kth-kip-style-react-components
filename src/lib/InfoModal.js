import React from 'react'
import { Button, Modal, ModalBody, ModalFooter } from 'reactstrap'
import ModalHeader from './ModalHeader'

const InfoModal = ({ id, isOpen, header, htmlBody, children, closeLabel, onToggle }) => {
  return (
    <Modal isOpen={isOpen} toggle={onToggle} id={id}>
      <ModalHeader header={header} >
        <button type='button' className='close' aria-label='Close' onClick={onToggle}>
          <span aria-hidden='true'>Ã—</span>
        </button>
      </ModalHeader>
      <ModalBody>
        {children}
        {htmlBody && <div dangerouslySetInnerHTML={{ __html: htmlBody }}></div>}
      </ModalBody>
      <ModalFooter>
        <Button color='secondary' onClick={onToggle}>
          {closeLabel}
        </Button>
      </ModalFooter>
    </Modal>
  )
}
export default InfoModal
