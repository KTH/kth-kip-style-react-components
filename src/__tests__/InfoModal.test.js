import React from 'react'
import renderer from 'react-test-renderer'
import InfoModal from '../lib/InfoModal'

const TEST_TRANSLATIONS = {
  header: 'Choose image',
  body: `Choose the image that will be displayed on the page Course information. 
    You can choose a default image based on the main subject of the course or choose to upload an image on your own choice. 
    The image will be displayed with the format 300px * 400px. The file format must be .png or .jpg.`,
  btnClose: 'Close'
}
let isOpen = true

xdescribe('InfoModal', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(<InfoModal isOpen={isOpen}
        modalId='infoPic' header={TEST_TRANSLATIONS.header} children={TEST_TRANSLATIONS.body} closeLabel={TEST_TRANSLATIONS.btnClose} />
      ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
