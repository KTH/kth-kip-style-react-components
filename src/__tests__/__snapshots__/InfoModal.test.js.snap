// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoModal renders properly 1`] = `
<div
  className="modal fade show"
  role="dialog"
  tabindex="-1"
  style="display: block;"
>
  <div
    id="infoPic"
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header h-4"
      >
        <h4
          className="modal-title"
        >
          Välj bild
        </h4>
        <button
          type="button"
          className="close"
          aria-label="Close"
        >
          <span
            aria-hidden="true">
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <p>
          Välj bild att visa på sidan Kursinformation. Du kan välja att visa en standardbild baserat på kursens huvudområde/ämne eller välja att ladda upp en egen bild. 
          Bilden kommer att visas med formatet 300px * 400px. Filformatet måste vara .png eller .jpg.
        </p>
      </div>
      <div
        className="modal-footer">
        <button
          type="button"
          className="btn btn-secondary"
        >
          Stäng
        </button>
      </div>
    </div>
  </div>
</div>
`;